application/src/test/java/org/thingsboard/server/service/sync/vc/VersionControlTest.java

```diff
@@ -562,6 +571,46 @@ public void testDefaultDeviceProfileVc_betweenTenants_findExisting() throws Exce
         checkImportedEntity(tenantId1, defaultDeviceProfile, tenantId2, importedDeviceProfile);
     }
 
+    @Test
+    public void testVcWithCalculatedFields_betweenTenants() throws Exception {
```

Comment:
Please add test covering circular references between CFs, e.g. device1 references asset1, and asset1 in turn references device1